chacractl/api/binaries.py

```diff
@@ -71,27 +86,44 @@ def post(self, url, filepath):
             return self.put(file_url, filepath)
         elif exists.status_code == 404:
             logger.info('POSTing file: %s', filepath)
-            with open(filepath, 'rb') as binary:
-                response = requests.post(
+            binary, digest = self.load_file(filepath)
+            response = requests.post(
                     url,
                     files={'file': binary},
                     auth=chacractl.config['credentials'],
                     verify=chacractl.config['ssl_verify'])
         if response.status_code > 201:
             logger.warning("%s -> %s", response.status_code, response.text)
             response.raise_for_status()
+        if not self.verify_upload(url, filename, digest):
+            # Since this is a new file, attempt to delete it
+            logging.error(
+                    'Checksum mismatch: server has wrong checksum for %s! '
+                    'Deleting corrupted file from server...',
+                    filepath)
+            self.delete(file_url)
+            raise SystemExit(
```

Comment:
if the client is uploading several files this would stop at the first occurrence. What would the problem be to simply continue, or after deleting the remote binary it simply tries again?